### ISSUE TITLE
When starting a multi-container service with rancher-compose, some containers are removed on start

### ISSUE THREAD

**Issue by egnoriega** [rancher/rancher#3943]
I'm trying to start a complex set of containers, which includes a number of sidekicks as data-containers.  Some of the containers are removed without specific explanation (or I'm not sure why anyway). I'm include a copy of the rancher-compose file, if anyone sees something wrong.

```
input:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.scheduler.affinity:host_label: MIC-Use=TCP_Modeling
    io.rancher.service.hash: d017b580c35d7cfea45f285c241988ee723ad842
  image: snip/uiio_input
  volumes_from:
  - locking
input-helper:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.scheduler.affinity:host_label: MIC-Use=TCP_Modeling
    io.rancher.service.hash: 889200a89f80bdff88cff08eee9d80e9bab80ea9
  command:
  - param_usecase2.csv
  image: snip/uiio_input-helper
  volumes_from:
  - locking
  - param
param:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.scheduler.affinity:host_label: MIC-Use=TCP_Modeling
    io.rancher.service.hash: cafe19ae9f2185d2c982db08085ff00381fb5e68
  command:
  - s3://mic.snip.org/tpc/param/param_usecase2.csv
  image: snip/uiio_param
  volumes_from:
  - locking
locking:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.scheduler.affinity:host_label: MIC-Use=TCP_Modeling
    io.rancher.service.hash: 355d70d20f994df3897b9c4ddf4cd2aa4d9b92ff
  image: busybox
  volumes:
  - /locks
modelrun:
  labels:
    io.rancher.sidekicks: locking,input,output,param,input-helper
    io.rancher.container.pull_image: always
    io.rancher.container.start_once: 'true'
    io.rancher.scheduler.affinity:host_label: MIC-Use=TCP_Modeling
    io.rancher.service.hash: 4b53ebdcc556a726f33d117e18c001abb2151938
  command:
  - mic-iloh33yz
  - '900'
  - '2015'
  - '2024'
  cpu_shares: 25
  image: snip/micgeneric:case2
  volumes_from:
  - input
  - output
  - param
  - locking
  mem_limit: 1000
output:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.scheduler.affinity:host_label: MIC-Use=TCP_Modeling
    io.rancher.service.hash: 50d7a4c346440407dddbde85cc9be9464f10b91a
  command:
  - s3://mic.snip.org/tpc/results/mic-iloh33yz/
  image: snip/uiio_output
  volumes_from:
  - locking
```

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v0.59.1` Cattle: `v0.148.1` UI: `v0.90.0` |
|Access|`localauth` `admin`|
|Route|`null`|

**Comment by deniseschannon** [comment]
Which containers are being removed? It is the container or service?

**Comment by egnoriega** [comment]
Hi Denise. It's the containers labeled output, param and input-helper; he
get added, but then are removed.  They show up under info with zero
containers.  The other containers spin up normally (...eventually they give
up and exit).

This is what is seen during service activation; the last three tabs are the
same:


​

**Comment by deniseschannon** [comment]
Would you be able to try using v0.63.0? We introduced an error state and gives users a lot more information about what could have gone wrong. I have tried using your docker-compose.yml in a simpler form (no scheduling rules and deleted some commands). 

Quick note on your docker-compose.yml. Since you have a sidekick on all your services, you only need the scheduling rule on the main service `modelrun` as all sidekicks will be scheduled on the same host as the primary service (the one with the sidekick label, aka `modelrun`). 
http://docs.rancher.com/rancher/rancher-compose/#sidekicks

My docker-compose.yml
```
input:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.service.hash: 16a858cd4f51bf179b42e0fcd59e9e7aa40ce76f
  image: snip/uiio_input
  volumes_from:
  - locking
input-helper:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.service.hash: 849fbc3e5a8dddbea1d2d3675e7960d6bdc96a8c
  image: snip/uiio_input-helper
  volumes_from:
  - locking
  - param
param:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.service.hash: bedea638731ab6952ae68a8a7f1e1f93a3fe7b0c
  image: snip/uiio_param
  volumes_from:
  - locking
locking:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.service.hash: 281aa259ac0f8da57a196dcc19bf6b2acc822c2d
  image: busybox
  volumes:
  - /locks
modelrun:
  labels:
    io.rancher.sidekicks: locking,input,output,param,input-helper
    io.rancher.container.start_once: 'true'
    io.rancher.service.hash: b2969ed1c18a455cdb6ce531bea86bd7fb9cad5a
  command:
  - mic-iloh33yz
  - '900'
  - '2015'
  - '2024'
  cpu_shares: 25
  image: snip/micgeneric:case2
  volumes_from:
  - input
  - output
  - param
  - locking
  mem_limit: 1000
output:
  labels:
    io.rancher.container.start_once: 'true'
    io.rancher.service.hash: 264dacbb45eac5b5abdd8d4ec6de7537ea36b0f3
  image: snip/uiio_output
  volumes_from:
  - locking
```

Error fails on input, so it can't continue to try and make the other containers.  Did your service actually get to Active state?
```
Image [snip/uiio_input:latest] failed to pull: Error: image snip/uiio_input not found
```

**Comment by egnoriega** [comment]
Ok, well more information:  I have been able to get the following message back: 
` Removed (Image [snip/uiio_output:latest] failed to pull: Error: Status 403 trying to pull repository snip/uiio_output: "\"The Repository is Locked, access denied\"")
Host:`

So it looks like Docker HUB is the central problem at this point. (I assumed it was a rancher issue since I could not get a clear message and the containers oddly appear under the standalone list in infrastructure.)  I'll follow up with Docker and let you know any resolution.

**Comment by egnoriega** [comment]
Yup, it turns out to be Docker HUB locking the repo.  There is a timing issue if you run low on private repos -- I deleted the repo, created it manually, and pushed again.  The containers start correctly. I've removed the extra scheduling labels just to clean things up.  The main process seems to restart even though I'm asking it not to, but that's a different problem.  Thanks much for your time on this, it's appreciated.
